pkg/daemon/config_drift_monitor_test.go

```diff
@@ -0,0 +1,414 @@
+						{
+							Contents: helpers.StrToPtr("unittest-service-contents"),
+							Name:     "10-unittest-service.conf",
+						},
+						{
+							Name: "20-unittest-service.conf",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// Determines which file to mutate and runs the appropriate mutator.
+func (tc configDriftMonitorTestCase) mutate(ignConfig ign3types.Config) error {
+	if tc.mutateFile != nil {
+		return tc.mutateFile(ignConfig.Storage.Files[0].Path)
+	}
+
+	if tc.mutateDropin != nil {
+		dropinPath := getIgn3SystemdDropinPath(tc.systemdPath, ignConfig.Systemd.Units[0], ignConfig.Systemd.Units[0].Dropins[0])
+		return tc.mutateDropin(dropinPath)
+	}
+
+	if tc.mutateUnit != nil {
+		unitPath := getIgn3SystemdUnitPath(tc.systemdPath, ignConfig.Systemd.Units[0])
+		return tc.mutateUnit(unitPath)
+	}
+
+	return fmt.Errorf("no file mutator provided")
+}
+
+// Creates and modifies the Ignition Config and converts it into a
+// MachineConfig suitable for this test.
+func (tc configDriftMonitorTestCase) getFixtures(t *testing.T) (ign3types.Config, *mcfgv1.MachineConfig) {
+	t.Helper()
+
+	ignConfig := tc.getIgnConfig()
+
+	// Prefix all the ignition files with the temp directory and write there.
+	for i, file := range ignConfig.Storage.Files {
+		file.Path = filepath.Join(tc.tmpDir, file.Path)
+		tc.writeFileForTest(t, file.Path, file.Contents.Source)
+		ignConfig.Storage.Files[i] = file
+	}
+
+	// Prefix all the systemd files with the temp dir and write them.
+	for _, unit := range ignConfig.Systemd.Units {
+		unitPath := getIgn3SystemdUnitPath(tc.systemdPath, unit)
+		tc.writeFileForTest(t, unitPath, unit.Contents)
+		for _, dropin := range unit.Dropins {
+			dropinPath := getIgn3SystemdDropinPath(tc.systemdPath, unit, dropin)
+			tc.writeFileForTest(t, dropinPath, dropin.Contents)
+		}
+	}
+
+	// Create a MachineConfig from our Ignition Config
+	mc := helpers.CreateMachineConfigFromIgnition(ignConfig)
+	mc.Name = "config-drift-monitor" + string(uuid.NewUUID())
+
+	return ignConfig, mc
+}
+
+func (tc configDriftMonitorTestCase) onDriftFunc(t *testing.T, err error) {
+	// If we're not expecting a configDriftErr, we should not end up here.
+	if tc.expectedErr == nil {
+		t.Errorf("expected no config drift error, but got one anyway: %s", err)
+	}
+
+	// Make sure that we get specific error types based upon the expected
+	// values
+	var cdErr configDriftErr
+	assert.ErrorAs(t, err, &cdErr)
+
+	// If the testcase asks for a fileConfigDriftErr, be sure we got one.
+	var fErr fileConfigDriftErr
+	if errors.As(tc.expectedErr, &fErr) {
+		assert.ErrorAs(t, err, &fErr)
+	}
+
+	// If the testcase asks for a unitConfigDriftErr, be sure we got one.
+	var uErr unitConfigDriftErr
+	if errors.As(tc.expectedErr, &uErr) {
+		assert.ErrorAs(t, err, &uErr)
+	}
+}
+
+func (tc configDriftMonitorTestCase) writeFileForTest(t *testing.T, path string, contents *string) {
+	t.Helper()
+	// I know this isn't how writeFiles actually does this, but I did it this way
```

Comment:
Same here. Try to reword comment without referring to "I" . This applies to other comments in the PR as well in case I missed some of them.